daemon/src/close_position/maker.rs

```diff
@@ -0,0 +1,250 @@
+    n_payouts: usize,
+}
+
+impl Actor {
+    pub fn new(executor: command::Executor, n_payouts: usize) -> Self {
+        Self {
+            tasks: Tasks::default(),
+            pending_protocols: HashMap::default(),
+            executor,
+            n_payouts,
+        }
+    }
+}
+
+#[async_trait]
+impl xtra::Actor for Actor {
+    type Stop = ();
+
+    async fn stopped(self) -> Self::Stop {}
+}
+
+#[xtra_productivity(message_impl = false)]
+impl Actor {
+    async fn handle(&mut self, msg: NewInboundSubstream, ctx: &mut xtra::Context<Self>) {
+        let NewInboundSubstream { peer, stream } = msg;
+        let address = ctx.address().expect("we are alive");
+
+        self.tasks.add_fallible(
+            async move {
+                let mut framed =
+                    Framed::new(stream, JsonCodec::<ListenerMessage, DialerMessage>::new());
+
+                let propose = framed
+                    .next()
+                    .await
+                    .context("End of stream while receiving Propose")?
+                    .context("Failed to decode Propose")?
+                    .into_propose()?;
+
+                address
+                    .send(ProposeReceived {
+                        propose,
+                        framed,
+                        peer,
+                    })
+                    .await?;
+
+                anyhow::Ok(())
+            },
+            |e| async move { tracing::warn!("Failed to handle incoming close position: {e:#}") },
+        );
+    }
+}
+
+#[xtra_productivity]
+impl Actor {
+    async fn handle(&mut self, msg: ProposeReceived) {
+        let ProposeReceived {
+            propose,
+            framed,
+            peer,
+        } = msg;
+        let order_id = propose.id;
+
+        let result = self
+            .executor
+            .execute(order_id, |cfd| {
+                // TODO: Validate that the correct peer is invoking this?
+                cfd.start_close_position_maker(propose.price, self.n_payouts, propose.unsigned_tx)
+            })
+            .await
+            .context("Failed to start close position protocol");
+
+        let (transaction, proposal) = match result {
+            Ok((transaction, proposal)) => (transaction, proposal),
+            Err(e) => {
+                tracing::debug!(%order_id, %peer, "Failed to start close position protocol: {e:#}");
+                emit_failed(order_id, e, &self.executor).await;
+                return;
+            }
+        };
+
+        self.pending_protocols
+            .insert(order_id, (framed, transaction, proposal, peer));
+    }
+
+    async fn handle(&mut self, msg: Accept) -> Result<()> {
+        let Accept { order_id } = msg;
+
+        let (mut framed, transaction, proposal, _peer) =
+            self.pending_protocols
+                .remove(&order_id)
+                .with_context(|| format!("No active protocol for order {order_id}"))?;
+
+        self.tasks.add_fallible(
```

Comment:
I think this part should be the same for both! That's the point of making it symmetric - the actual execution of the protocol (after accept) is the same for both. 
I imagine this to be similar to what we have in `rollover` where both dispatch to the same function. I think that's a good pattern to ensure symmetry.

Reply:
Not sure we are actually going for a symmetric protocol, please see https://github.com/itchysats/itchysats/pull/2135#issuecomment-1142868508